plugins/core-plugin/src/main/resources/Dockerfile-template-xlang

```
@@ -16,13 +14,13 @@ ${filesToCopy}
 ${directoriesToCopy}
 WORKDIR $WORKDIR
 
-# Create requirements.txt file if not provided
-RUN if ! [ -f requirements.txt ] ; then echo "$BEAM_PACKAGE" > requirements.txt ; fi
+# Throw if requirements.txt file not provided
+RUN if ! [ -f requirements.txt ] ; then >&2 echo "error: no requirements.txt file found" && exit 1 ; fi
 
 # Install dependencies to launch the pipeline and download to reduce startup time
 RUN python -m venv /venv \
-    && /venv/bin/pip install --no-cache-dir --upgrade pip setuptools \
-    && /venv/bin/pip install --no-cache-dir -U -r $REQUIREMENTS_FILE \
+    && python -m ensurepip --upgrade \
```

Does this actually ensure pip for the venv?
```suggestion
    && /venv/bin/python -m ensurepip --upgrade \
```

Same for the others

---

I added the suggestion, not sure I understand the question though.

---

"The ensurepip package provides support for bootstrapping the pip installer into an existing Python installation or virtual environment."

I just wasn't sure if calling `python -m ensurepip` would perform the intended function for the venv since it was running on the system python rather than the venv python (which I think is just a pointer to the system python, but again I was just unsure)